crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -347,7 +352,21 @@ fn convert_to_list_extend(
     let semantic = checker.semantic();
     let locator = checker.locator();
     let if_str = match if_test {
-        Some(test) => format!(" if {}", locator.slice(test.range())),
+        Some(test) => {
+            // If the test is an assignment expression,
+            // we must parenthesize it when it appears
+            // inside the comprehension to avoid a syntax error.
+            //
+            // Notice that we do not need `any_over_expr` here,
+            // since if the assignment expression appears
+            // internally (e.g. as an operand in a boolean
+            // operation) then it will already be parenthesized.
```

Comment:
This was my understanding of the last paragraph [here](https://docs.python.org/3/reference/expressions.html#assignment-expressions), but someone should check me.

Reply:
Yeah, I think that's correct. In any other position, the assignment expression should already surrounded by parentheses. There's the slice expression which doesn't requires parentheses because of the square brackets (`if data[x := 2]: ...`) but that shouldn't create a syntax error with the current autofix (https://play.ruff.rs/81286e9b-248e-4707-9abf-fdd6f38e95cb).